src/test/java/streams/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,131 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise1 {
+
+    @Test
+    public void findPersonsEverWorkedInEpam() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toList()
+        List<Person> personsEverWorkedInEpam = employees.stream()
+                .filter(employee -> employee.getJobHistory().stream()
+                        .anyMatch(entry -> "epam".equalsIgnoreCase(entry.getEmployer())))
+                .map(Employee::getPerson)
+                .collect(Collectors.toList());
+
+        List<Person> expected = Arrays.asList(
+                employees.get(0).getPerson(),
+                employees.get(1).getPerson(),
+                employees.get(4).getPerson(),
+                employees.get(5).getPerson());
+        assertEquals(expected, personsEverWorkedInEpam);
+    }
+
+    @Test
+    public void findPersonsBeganCareerInEpam() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toList()
+        List<Person> startedFromEpam = employees.stream()
+                .filter(employee -> employee.getJobHistory().stream()
+                        .findFirst()
+                        .get()
+                        .getEmployer()
+                        .equalsIgnoreCase("epam"))
+                .map(Employee::getPerson)
+                .collect(Collectors.toList());
+
+        List<Person> expected = Arrays.asList(
+                employees.get(0).getPerson(),
+                employees.get(1).getPerson(),
+                employees.get(4).getPerson());
+        assertEquals(expected, startedFromEpam);
+    }
+
+    @Test
+    public void findAllCompanies() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toSet()
+        Set<String> companies = employees.stream()
+                .map(Employee::getJobHistory)
+                .flatMap(Collection::stream)
+                .map(JobHistoryEntry::getEmployer)
+                .collect(Collectors.toSet());
+
+        Set<String> expected = new HashSet<>();
+        expected.add("EPAM");
+        expected.add("google");
+        expected.add("yandex");
+        expected.add("mail.ru");
+        expected.add("T-Systems");
+        assertEquals(expected, companies);
+    }
+
+    @Test
+    public void findMinimalAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация
+        Integer minimalAge = employees.stream()
+                .map(employee -> employee.getPerson().getAge())
+                .min(Integer::compareTo)
+                .get();
```

Comment:
It's better to make
```Java
.mapToInt(....getAge())
.min();
```
because no wrapping (getAge() returns `int`) and one less operation.